### DISCUSSION TITLE
Where to put code?

### DISCUSSION THREAD

**Discussion by sefnyn** [Digital-Preservation-Coalition/PythonStudyGroups#21] [Questions] [answered]
I'm in Group 15.  We decided our first task would be writing code to remove unwanted files.  We did not decide (yet) to put our code in a code repository.  What are other groups doing?

**Comment by harrylloyd-bl** [comment, marked as answer]
I'd definitely recommend trying to put it in a public repo together - it'll be a really good learning experience! One of you set up a repo on your github account, then go to settings/collaborators and you can add people by username/email. Once that's done you can all push things to the repository together! Maybe you want to all work on separate versions of the code to start with e.g. by adding your initials to the files 'remove_files_HL.py' but eventually you could build up to pull requests and really working together!

![image](https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/assets/125367849/672cdcc7-f94a-453b-8858-7d8c5b63271f)

**Comment by mutanthumb** [comment]
Hi, I also just made folders for each group: https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/tree/main/Cohort1 , you can add your code here if you'd like as well.

**Reply by JoCastle** [reply]
Hi, I've just tried to add a script to the Group 19 folder but am struggling to work out how to do that? When I click the 'Add file' button at the top-right of the screen I get a message saying:  Uploads are disabled. File uploads require push access to this repository. If I click on the + in the file explorer on the left it tells me I need to fork the repository before I can change anything.

This may be me not understanding how to use GitHub but is there something else I should be doing?

**Reply by mutanthumb** [reply]
Hi @JoCastle 
I'm sorry I missed this last week. I just added you officially as an outside contributor with "write" access so you should be able to add files now. Let me know if you still can't add files. 
Thanks,
susan
